### ISSUE TITLE
CI fails on Windows

### ISSUE THREAD

**Issue by t-imamichi** [Qiskit/qiskit-optimization#241]
### Information

- **Qiskit Optimization version**: latest
- **Python version**: 3.8
- **Operating system**: Windows 10

### What is the current behavior?

CI fails. https://github.com/Qiskit/qiskit-optimization/actions/runs/1178358017

```
test.algorithms.test_recursive_optimization.TestRecursiveMinEigenOptimizer.test_recursive_warm_qaoa
---------------------------------------------------------------------------------------------------

Captured traceback:
~~~~~~~~~~~~~~~~~~~
    Traceback (most recent call last):

      File "D:\a\qiskit-optimization\qiskit-optimization\test\optimization_test_case.py", line 47, in wrapper
    test_item(self, *args)

      File "D:\a\qiskit-optimization\qiskit-optimization\test\algorithms\test_recursive_optimization.py", line 155, in test_recursive_warm_qaoa
    np.testing.assert_array_almost_equal(cplex_result.x, result.x, 4)

      File "c:\miniconda\lib\site-packages\numpy\testing\_private\utils.py", line 1046, in assert_array_almost_equal
    assert_array_compare(compare, x, y, err_msg=err_msg, verbose=verbose,

      File "c:\miniconda\lib\site-packages\numpy\testing\_private\utils.py", line 844, in assert_array_compare
    raise AssertionError(msg)

    AssertionError: 
Arrays are not almost equal to 4 decimals

Mismatched elements: 2 / 2 (100%)
Max absolute difference: 2.
Max relative difference: 1.
 x: array([0., 2.])
 y: array([2., 0.])
```

**Comment by woodsp-ibm** [comment]
In looking at the test case, from what I can see the problem is symmetric, in the two variables, such that the solution found is correct, just it has swapped the variables
```
Minimize
 obj1: 2 x0 + 2 x1 + [ 4 x0 * x0 + x1 * x1 + 0.5 x0 * x1 + 0.5 x0 * x1 ] / 2
Subject To
 constraint: x0 + x1  = 2
Bounds
 0 <= x0 <= 4
 0 <= x1 <= 4
 ```
 Hence to me it seems that the test case could be improved to cover that - given that this seems most likely to be platform dependent behavior in arriving at the solution.

**Comment by t-imamichi** [comment]
Good catch. It might be a reason. Unfortunately, this issue is hard for me to resolve because it does not reproduce on my local Windows 10 environment. All unit tests pass successfully...

**Comment by woodsp-ibm** [comment]
Your Win 10 env is with Python 3.8 like the unit test? I can see if I can reproduce it locally (I have 3.8.8). Either way I think this is a valid result right and if so then the fix just seems to be to allow either solution as a valid result in which case CI should then pass, even if in some other env we cannot reproduce the exact solution it ends up with in CI.

**Comment by t-imamichi** [comment]
Yes, I use Python 3.8.10 on my Windows 10. I'm wondering what's the difference of my env and (your env or CI env). If it reproduces on my env, I can work on it...

**Comment by woodsp-ibm** [comment]
I have not been able to reproduce the error yet. I made a script from the test case and have it in a for loop. It was running and I could not reproduce the failure and I was looking at Cplex and saw it has a `randomseed` parameter so stopped the test and  altered the CplexOptimizer creation to pass that as a cplex_parameter - plus for each loop iteration I made it a random value. It ran earlier for 10k iterations without failing - so either I am not able to reproduce it either or its a very slim chance of it failing and I did not see it yet. I started the script again since I had the loop set to end at 10k iterations - I increased that number and figured I'd leave it run overnight and see. So far its still going.
